conformance/tests/gatewayclass-supported-version-condition.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	apiextv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+	"sigs.k8s.io/gateway-api/pkg/consts"
+	"sigs.k8s.io/gateway-api/pkg/features"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, GatewayClassSupportedVersionCondition)
+}
+
+var GatewayClassSupportedVersionCondition = suite.ConformanceTest{
+	ShortName: "GatewayClassSupportedVersionCondition",
+	Features: []features.FeatureName{
+		features.SupportGateway,
+	},
+	Description: "A GatewayClass should set the SupportedVersion condition based on the presence and version of Gateway API CRDs in the cluster",
```

Comment:
I'm not sure this makes sense. 

Consider I build an implementation that supports v1.1. 

v1.2 comes out. Per the spec, even if I am willing to support it, I must set the condition to "false" 

This means I cannot pass this conformance test for 1.2 until I upgrade to 1.2 which doesn't actually seem desired?